build

```diff
@@ -97,25 +95,26 @@ elif [ -n "$TARGET" ]; then
 fi
 
 
-build_args=""
+build_args=()
+
+while read -r line && test -n "${line}"; do
+  build_args+=('--build-arg')
+  build_args+=("${line}")
+done <<<"$(env | grep -E '^BUILD_ARG_' | sed -n 's/^BUILD_ARG_//p')"
+
 if [ -n "$BUILD_ARGS_FILE" ]; then
   if [ ! -f "$BUILD_ARGS_FILE" ]; then
     echo "build_args file '$BUILD_ARGS_FILE' does not exist"
     exit 1
   fi
 
-  expanded_build_args=()
-  while IFS= read -r line || [ -n "$line" ];
+  while IFS= read -r line && test -n "${line}" || [ -n "$line" ];
```

Comment:
`test -n "${line}" || [ -n "$line" ]` are both the same thing (`[` is an alias to `test`):

```
alex@wx020:~/workspace/builder-task$ help [
[: [ arg... ]
    Evaluate conditional expression.
    
    This is a synonym for the "test" builtin, but the last argument must
    be a literal `]', to match the opening `['.
```